### PR TITLE
Add convenience constructor to InletVector and make its single-arg ctors explicit

### PR DESCRIPTION
**Pull request by joshessman-llnl**
# Summary

Follow-up to #686

### CODE REVIEW

**Review by white238 — APPROVED** [review]

**Comment by white238** [comment]
Thank you for cleaning this up quickly @joshessman-llnl

File: src/axom/inlet/examples/mfem_coefficient.cpp [hunk 1]
```diff
@@ -100,7 +100,7 @@ struct BoundaryCondition
 
 inlet::InletVector toInletVector(const mfem::Vector& vec)
 {
-  return {axom::primal::Vector3D {vec.GetData(), vec.Size()}, vec.Size()};
+  return inlet::InletVector {vec.GetData(), vec.Size()};
```
Comment (kennyweiss) [inline src/axom/inlet/examples/mfem_coefficient.cpp:103]: 👍

**Review by kennyweiss — APPROVED** [review]
Thanks @joshessman-llnl

**Comment by kennyweiss** [comment]
Partially addresses #693

**Pull request merged**